### ISSUE TITLE
The effect of valid_mask?

### ISSUE THREAD

**Issue by ChunmengLiu1** [chenqi1126/SIPE#12]
Hi! Thanks for the great jod!
When I read the code, I was confused about what valid_mask does. From data_voc.py I can see that it is a matrix with partial values of 1 (16, 21, h, w) determined from random cropping. However, in resnet50_SIPE.py, 
**norm_cam=F.interpolate(norm_cam, side3.shape[2:], mode='bilinear', align_corners=True)*valid_mask.**
What does this code mean?

**Comment by chenqi1126** [comment]
Hi @xinyuaning, 
Thanks for your attention. Since Random cropping operation introduces extra padding outside the original image extent, we treat these padding areas as invalid. So 'valid_mask' is used to 1) eliminate irrelevant responses in localization maps, and 2) ensure GSC loss actioned on valid areas.

**Comment by ChunmengLiu1** [comment]
Thank you for your timely response!
